utils/build-script-impl

```diff
@@ -308,6 +309,51 @@ function float_min() {
     fi
 }
 
+# Support for performing isolated actions.
+#
+# This is part of refactoring more work to be done or controllable via
+# `build-script` itself. For additional information, see:
+# https://bugs.swift.org/browse/SR-237
+#
+# To use this functionality, the script is invoked with:
+#   ONLY_EXECUTE=<action name>
+# where <action name> is one of:
+#   all                          -- execute all actions
+#   ${host}-${product}-build     -- the build of the product
+#   ${host}-${product}-test      -- the test of the product
+#   ${host}-${product}-install   -- the install of the product
+#   ${host}-package              -- the package construction and test
+#   merged-hosts-lipo            -- the lipo step, if used
+# and if used, only the one individual action will be performed.
+#
+# If not set, the default is `all`.
+
+# should_execute_action(name) -> 1 or nil
+#
+# Check if the named action should be run in the given script invocation.
+function should_execute_action() {
+    local name="$1"
+    
+    if [[ "${ONLY_EXECUTE}" = "all" ]] ||
+           [[ "${ONLY_EXECUTE}" = "${name}" ]]; then
+        echo 1
+    fi
+}
+
+# should_execute_host_actions_for_phase(host, phase-name) -> 1 or nil
+#
+# Check if the there are any actions to execute for this host and phase (i.e.,
+# "build", "test", or "install")
+function should_execute_host_actions_for_phase() {
```

Comment:
Thank you for moving this into a function!  The old logic seemed simple, but it turned out to be somehow confusing for me.  With a function it is much more clear what is happening.